jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/CachedJMSConnectionFactory.java

```diff
@@ -129,14 +112,29 @@ public Connection getConnection(String userName, String password) throws JMSExce
     @Override
     public Connection createConnection(String userName, String password) throws JMSException {
         Connection connection;
-        if (userName == null || password == null) {
-            connection = super.createConnection();
+        if (userName == null && password == null) {
+            connection = createConnection();
         } else {
-            connection = super.createConnection(userName, password);
-        }
-        if (this.cacheLevel >= JMSConstants.CACHE_CONNECTION) {
-            cachedConnection = connection;
+            if (cachedConnection == null) {
+                connection = super.createConnection(userName, password);
+                cachedConnection = connection;
+            } else {
+                connection = cachedConnection;
+            }
+
+            try {
+                connection.start();
```

Comment:
A call to Connection.start on a connection that has already been started is ignored.